ts: Generate sp_layout.json after SP builds

imre-kis-arm opened the pull request:
TF-A requires to have the SPs in the same order in the FIP as it is defined in ARM_BL2_SP_LIST_DTS. The sp_layout.json controls the order of SPs in the FIP package. Previously the individual JSON fragments were appended to the sp_layout.json file at the end of the SP build. The final order of SP definitions in the sp_layout.json depended on the SP build order which was non-deterministic because of the parallel make jobs. To resolve this issue the JSON merging step is now done after all SPs have been built. The order will follow the order of build-sp calls in the fvp-psa-sp.mk which matches the order of SPs in ARM_BL2_SP_LIST_DTS.

<!--
    If you are new to submitting pull requests to OP-TEE, then please have a
    look at the list below and tick them off before submitting the pull request.

    1. Read our contribution guidelines:
         https://optee.readthedocs.io/en/latest/general/contribute.html

    2. Read the contribution section in Notice.md and pay extra attention to the
       "Developer Certificate of Origin" in the contribution guidelines.

    3. You should run checkpatch preferably before submitting the pull request.

    4. When everything has been reviewed, you will need to squash, rebase and
       add tags like `Reviewed-by`, `Acked-by`, `Tested-by` etc. More details
       about this can also be found on the link provided above.

    NOTE: This comment will not be shown in the pull request, so no harm keeping
    it, but feel free to remove it if you like.
-->

jforissier reviewed (commented):
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

jenswi-linaro commented on trusted-services.mk:
```diff
@@ -78,12 +78,18 @@ ffa-sp-all-clean: ffa-$1-sp-clean
+$(TS_INSTALL_PREFIX)/opteesp/json/sp_layout.json: ffa-sp-all
+	python $(TS_PATH)/tools/python/merge_json.py $@ $(fip_sp_json_paths)
+
+optee-os-common: $(TS_INSTALL_PREFIX)/opteesp/json/sp_layout.json
```
This is a strange dependency. I can understand the `optee-os-common: ffa-sp-all` dependency above to avoid making changes in the main make file. However, this dependency is even broken, I assume that `ffa-sp-all` must complete before the `sp_layout.json` file can be generated.

jforissier replied:
Isn't it the purpose of the recipe just above? (although the command does not contain the target which is ugly but I assume the Python script has a hard-coded output path?)

jenswi-linaro replied:
You're right, I missed that.

jenswi-linaro commented:
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

imre-kis-arm commented:
I've added the Acked-by tags and I also made a slight modification in the `sp_layout.json`'s path. I moved it to `$(TS_INSTALL_PREFIX)` from `$(TS_INSTALL_PREFIX)/opteesp/json` because in case of a raw binary SPs, trusted-services would use `$(TS_INSTALL_PREFIX)/sp` as it's install directory and it would be misleading to place the `sp_layout.json` into the `opteesp` directory.
Please let me know if this is fine and the Acked-by tags are still stand for the PR.

imre-kis-arm commented:
Thank you!
